client-js/app/elements/labrad-nodes.ts

```diff
@@ -626,47 +636,43 @@ export class LabradNodes extends polymer.Base {
   }
 
 
-  private updateServerAutostartStatus(serversName: string,
-                                      serverIdx: number): void {
-    const server = this[serversName][serverIdx];
-    server.autostart = false;
-    for (const node of server.nodes) {
-      if (node.exists && node.autostart) {
-        server.autostart = true;
-        return;
+  private updateServerVersion(listName: string, serverIdx: number): void {
+    const server = this[listName][serverIdx];
+    for (let nodeIdx = 0; nodeIdx < server.nodes.length; nodeIdx++) {
+      const node = server.nodes[nodeIdx];
+      if (node.exists) {
+        const status = this.getServerStatus(node.name, server.name);
+        const version = status.outdated
+                      ? `running=${status.runningVersion}, latest=${status.latestVersion}`
+                      : status.version;
+        if (node.version !== version) {
+          this.set(`${listName}.#${serverIdx}.nodes.#${nodeIdx}.version`,
+                   version);
```

Comment:
That doesn't work for changing properties on the elements within the arrays, which is what we're doing here. Also, if I'm reading the [docs](https://www.polymer-project.org/2.0/docs/api/elements/Polymer.Element#method-notifySplices) right we have to provide the `notifySplices` method with a list of splice objects describing the changes, so it's probably easier to just use the polymer splice methods to modify the arrays in the first place.